clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -579,11 +629,34 @@ private void handleListOffsetResponse(Map<TopicPartition, Long> timestampsToSear
             ListOffsetResponse.PartitionData partitionData = listOffsetResponse.responseData().get(topicPartition);
             Errors error = Errors.forCode(partitionData.errorCode);
             if (error == Errors.NONE) {
-                OffsetAndTimestamp offsetAndTimestamp = null;
-                if (partitionData.offset != ListOffsetResponse.UNKNOWN_OFFSET)
-                    offsetAndTimestamp = new OffsetAndTimestamp(partitionData.offset, partitionData.timestamp);
-                log.debug("Fetched {} for partition {}", offsetAndTimestamp, topicPartition);
-                timestampOffsetMap.put(topicPartition, offsetAndTimestamp);
+                if (partitionData.offsets != null) {
```

Comment:
I think AbstractResponse should have a version number in it, since the data it contains varies based on the version.  Sometimes the interpretation of the data varies based on version as well.  It's a property of the response and it belongs in the response object.  But hopefully we can hold off on that for now since this patch is kinda big as-is...